source/exe/process_wide.cc

```diff
@@ -0,0 +1,33 @@
+#include "exe/process_wide.h"
+
+#include "common/common/assert.h"
+#include "common/event/libevent.h"
+#include "common/http/http2/nghttp2.h"
+
+#include "server/proto_descriptors.h"
+
+#include "ares.h"
+
+#ifdef ENVOY_GOOGLE_GRPC
+#include "grpc/grpc.h"
+#endif
+
+namespace Envoy {
+
+CleanupPtr ProcessWide::setup() {
```

Comment:
Would it be worthwhile to have some sort of check that this function is only called once, or (preferably) that calls to it are matched with interleaved cleanup calls (for testing contexts)?